### PR TITLE
i#3544 riscv64, part 3: add instruction set listings

### PR DESCRIPTION
**Pull request by semihalf-kardach-stanislaw**
Add files describing RISC-V instruction encodings for the base spec and each currently ratified extension. Those files are used as a data source by the codec.py generator script.

Currently the prefetch.[irw] instructions (part of Zicbop extension) are commented out as they cause a bug in the trie generation algorithm (due to aliasing encoding of prefetch.[irw] and ori).

Issue #3544

Signed-off-by: Stanislaw Kardach <kda@semihalf.com>

### CODE REVIEW

File: core/ir/riscv64/isl/README.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+If you want to add a new field:
+
+1. In `codec.py`:
+
+    1. Add new value to the `Field` enum.
+    2. If this is a destination field, update the `Field.is_dest()` function.
+    3. Update stringification functions: `Field.arg_name()` for C-macro argument
+       name and `Field.asm_name()` for assembly syntax comment override (if
+       necessary).
+    4. Add a new entry to the `cmt_str` list in `Field.arg_comment()` used as a
```
Comment (abhinav92003) [inline core/ir/riscv64/isl/README.md:126]: Having cmt_str and sz_str separate from the fields is a bit confusing and error-prone. Can we combine these three somehow?

File: core/ir/riscv64/isl/README.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
+# Instruction Set Listing
+
+## Format
+
+```text
+<name> | <instruction-format> | [field] | <fixedbits>
```
[the remark this answers is not in the document]
Comment (semihalf-kardach-stanislaw) [reply core/ir/riscv64/isl/README.md:6]: Instruction mnemonic to be precise.

File: core/ir/riscv64/isl/README.md (hunk 1 above)
Comment (semihalf-kardach-stanislaw) [reply core/ir/riscv64/isl/README.md:126]: Good idea. I'll combine those.

File: core/ir/riscv64/isl/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (semihalf-kardach-stanislaw) [reply core/ir/riscv64/isl/README.md]: Done.

File: core/ir/riscv64/isl/README.md (hunk 1 above)
Comment (semihalf-kardach-stanislaw) [reply core/ir/riscv64/isl/README.md]: Done. I've bundled all this info into Field object attributes. This makes it really hard to not fill all the required attributes when adding a new field.

**Review by abhinav92003 — APPROVED** [review]

**Pull request merged**